agent/testagent.go

```diff
@@ -274,7 +216,6 @@ func (a *TestAgent) Start(t *testing.T) (err error) {
 	a.Agent.StartSync()
 
 	if err := a.waitForUp(); err != nil {
-		cleanupTmpDir()
 		a.Shutdown()
```

Comment:
In a follow up we could register this `Shutdown` using `t.Cleanup` as well, and remove the need to call it in every test.